activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/InclusiveGatewayActivityBehavior.java

```diff
@@ -69,33 +69,37 @@ protected void executeInclusiveGatewayLogic(ExecutionEntity execution) {
 
     Collection<ExecutionEntity> allExecutions = executionEntityManager.findChildExecutionsByProcessInstanceId(execution.getProcessInstanceId());
     Iterator<ExecutionEntity> executionIterator = allExecutions.iterator();
-    boolean oneExecutionCanReachGateway = false;
-    while (!oneExecutionCanReachGateway && executionIterator.hasNext()) {
+    boolean oneExecutionCanReachGatewayInstance = false;
+    while (!oneExecutionCanReachGatewayInstance && executionIterator.hasNext()) {
       ExecutionEntity executionEntity = executionIterator.next();
       if (!executionEntity.getActivityId().equals(execution.getCurrentActivityId())) {
-        boolean canReachGateway = ExecutionGraphUtil.isReachable(execution.getProcessDefinitionId(), executionEntity.getActivityId(), execution.getCurrentActivityId());
-        if (canReachGateway) {
-          oneExecutionCanReachGateway = true;
-        }
-      } else if (executionEntity.getActivityId().equals(execution.getCurrentActivityId()) && executionEntity.isActive()) {
+          if (ExecutionGraphUtil.isReachable(execution.getProcessDefinitionId(), executionEntity.getActivityId(), execution.getCurrentActivityId())) {
+              //Now check if they are in the same "execution path"
```

Comment:
Clean coding: instead of writing a comment to explain what the code is doing it's better to extract a method with a meaningful name. For instance, you can extract a method `isSameExecutionPath` or `hasSameParent` and use it in the `if` condition. Also as Codacy is suggesting below the two `if` blocks could be combined.

```java
        if (isSameExecutionPath(execution, executionEntity) && 
              ExecutionGraphUtil.isReachable(execution.getProcessDefinitionId(), executionEntity.getActivityId(), execution.getCurrentActivityId())) {
              oneExecutionCanReachGatewayInstance = true;
          }
```
```java
    private boolean isSameExecutionPath(ExecutionEntity gatewayExecution, ExecutionEntity activeExecution) {
        return activeExecution.getParentId().equals(gatewayExecution.getParentId());
    }
```

Reply:
Done as suggested